lib/npo.src.js

```diff
@@ -77,6 +124,15 @@
 			cycle = timer(scheduling_queue.drain);
 		}
 	}
+	
+	function hasCatch(chain) {
+		for (var i = 0; i < chain.length; i++) {
+			if (chain[i].failure) {
+				return true;
+			}
+		}
+		return false;
+	}
```

Comment:
Yeah, current behavior is wrong I believe.

```js
var A = Promise.reject("oops");
var B = A.then(function(){});
var C = B.catch(function(){});
```

With NPO, I believe this would currently report an uncaught exception (maybe 2!), but with chrome, no exception reported.